ocs_ci/ocs/ripsaw.py

```diff
@@ -0,0 +1,76 @@
+
+"""
+RipSaw Class to run various workloads and scale tests
+"""
+import logging
+import tempfile
+
+from ocs_ci.ocs.exceptions import CommandFailed
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.utility.utils import run_cmd
+
+log = logging.getLogger(__name__)
+
+
+class RipSaw(object):
+    """
+      Workload operation using RipSaw
+    """
+
+    def __init__(self, **kwargs):
+        """
+        Initializer function
+
+        Args:
+            kwargs (dict):
+                Following kwargs are valid
+                repo: Ripsaw repo to used - a github link
+                branch: branch to use from the repo
+                namespace: namespace for the operator
+
+        Example Usage:
+            r1 = RipSaw()
+            r1.apply_crd(crd='ripsaw_v1alpha1_ripsaw_crd.yaml')
+            # use oc apply to apply custom modified bench
+            my_custom_bench = my_custom_bench.yaml
+            run_cmd('oc apply -f my_custom_bench')
+        """
+        self.args = kwargs
+        self.repo = self.args.get('repo', 'https://github.com/cloud-bulldozer/ripsaw')
+        self.branch = self.args.get('branch', 'master')
+        self.namespace = self.args.get('namespace', 'ripsaw')
+        self.ocp = OCP()
+        # self._create_namespace()
+        self._clone_ripsaw()
+
+    def _create_namespace(self):
+        """
+        create namespace for RipSaw
+        """
+        self.ocp.new_project(self.namespace)
+
+    def _clone_ripsaw(self):
+        """
+        clone the ripaw repo
+        """
+        self.dir = tempfile.TemporaryDirectory(prefix='ripsaw')
+        try:
+            cd_cmd = f'cd {self.dir.name} ; '
+            git_clone_cmd = f'git clone -b {self.branch} {self.repo} '
+            run_cmd(f'{cd_cmd} {git_clone_cmd}')
+            self.crd = 'resources/crd/'
+            self.operator = 'resources/operator.yaml'
+        except CommandFailed as cf:
+            log.error('Error during cloning of ripsaw repository')
+            raise cf
+
+    def apply_crd(self, crd):
```

Comment:
can you please add a usage example in the docstring? 
Also, don't we need run() and setup() methods as in fio.py?

Reply:
the operator/crd handles it in async way, so we have to add methods to parse the logs properly, those methods will be added when we get better understanding of what to gather and fail on.